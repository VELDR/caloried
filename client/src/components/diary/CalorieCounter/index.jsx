import PropTypes from 'prop-types';
import { useState } from 'react';
import { CircularProgressbarWithChildren, buildStyles } from 'react-circular-progressbar';
import 'react-circular-progressbar/dist/styles.css';
import caloriesIcon from '@static/images/calories.svg';

import classes from './style.module.scss';

const CalorieCounter = ({ caloriesConsumed, dailyCalorieGoal }) => {
  const [isHovered, setIsHovered] = useState(false);

  const remainingCalories = dailyCalorieGoal - caloriesConsumed;
  const percentage = (caloriesConsumed / dailyCalorieGoal) * 100;

  const handleMouseEnter = () => setIsHovered(true);
  const handleMouseLeave = () => setIsHovered(false);

  return (
    <div className={classes.container} onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
      <CircularProgressbarWithChildren
        value={percentage}
        circleRatio={0.75}
        styles={buildStyles({
          rotation: 1 / 2 + 1 / 8,
          strokeLinecap: 'butt',
          pathTransitionDuration: 0.5,
          pathColor: `var(--color-primary)`,
          trailColor: 'var(--color-bg-secondary)',
          backgroundColor: 'var(--color-bg-primary)',
        })}
      >
        {isHovered ? (
          <div className={classes.content}>
            <img src={caloriesIcon} alt="" className={classes.icon} />
            <div className={classes.hovered}>
              {caloriesConsumed}/{dailyCalorieGoal}
            </div>
          </div>
        ) : (
          <div className={classes.content}>
            <div className={classes.title}>
              {remainingCalories}
              <span>kcal</span>
            </div>
            <div className={classes.subtitle}>Remaining</div>
          </div>
        )}
      </CircularProgressbarWithChildren>
    </div>
  );
};

CalorieCounter.propTypes = {
  caloriesConsumed: PropTypes.number,
  dailyCalorieGoal: PropTypes.number,
};

export default CalorieCounter;
