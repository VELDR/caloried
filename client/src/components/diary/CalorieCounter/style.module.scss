@import '@styles/base';

.container {
  @include box(9rem);
  .content {
    @include flex-center-column;

    .icon {
      @include box(2rem);
    }

    .hovered {
      font-weight: 500;
      font-size: $font-size-medium;
    }
    .title {
      cursor: pointer;
      font-size: $font-size-larger;
      font-weight: 500;
      span {
        font-size: $font-size-small;
        font-weight: 300;
      }
    }
    .subtitle {
      text-transform: uppercase;
      font-size: $font-size-small;
      font-weight: 300;
    }
  }
}
