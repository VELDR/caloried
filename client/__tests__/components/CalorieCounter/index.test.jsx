import CalorieCounter from '@components/diary/CalorieCounter';
import { fireEvent, render } from '@utils/testHelper';

const mockProps = {
  caloriesConsumed: 1500,
  dailyCalorieGoal: 2000,
};

let wrapper;

beforeEach(() => {
  wrapper = render(<CalorieCounter {...mockProps} />);
});

describe('CalorieCounter Component', () => {
  it('should render component correctly', () => {
    const { getByTestId } = wrapper;

    const container = getByTestId('calorie-counter-container');
    expect(container).toBeInTheDocument();
  });

  it('should display consumed and goal calories on hover', () => {
    const { getByTestId, getByText } = wrapper;

    const container = getByTestId('calorie-counter-container');

    fireEvent.mouseEnter(container);
    expect(getByText(`${mockProps.caloriesConsumed}/${mockProps.dailyCalorieGoal}`)).toBeInTheDocument();

    fireEvent.mouseLeave(container);
    expect(getByText(`${mockProps.dailyCalorieGoal - mockProps.caloriesConsumed}`)).toBeInTheDocument(); // Adjust based on the actual expected content
  });

  it('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
