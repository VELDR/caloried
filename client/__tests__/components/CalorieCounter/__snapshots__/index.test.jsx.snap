// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalorieCounter Component should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
        data-testid="calorie-counter-container"
      >
        <div
          data-test-id="CircularProgressbarWithChildren"
        >
          <div
            style="position: relative; width: 100%; height: 100%;"
          >
            <svg
              class="CircularProgressbar "
              data-test-id="CircularProgressbar"
              viewBox="0 0 100 100"
            >
              <path
                class="CircularProgressbar-trail"
                d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
                fill-opacity="0"
                stroke-width="8"
                style="stroke: var(--color-bg-secondary); stroke-linecap: butt; transform: rotate(0.625turn); transform-origin: center center; stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 72.25663103256524px;"
              />
              <path
                class="CircularProgressbar-path"
                d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
                fill-opacity="0"
                stroke-width="8"
                style="stroke: var(--color-primary); stroke-linecap: butt; transform: rotate(0.625turn); transform-origin: center center; transition-duration: 0.5s; stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 126.44910430698917px;"
              />
            </svg>
            <div
              data-test-id="CircularProgressbarWithChildren__children"
              style="position: absolute; width: 100%; height: 100%; margin-top: -100%; display: flex; flex-direction: column; justify-content: center; align-items: center;"
            >
              <div
                class="content"
              >
                <div
                  class="title"
                >
                  500
                  <span>
                    kcal
                  </span>
                </div>
                <div
                  class="subtitle"
                >
                  Sisa
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      data-testid="calorie-counter-container"
    >
      <div
        data-test-id="CircularProgressbarWithChildren"
      >
        <div
          style="position: relative; width: 100%; height: 100%;"
        >
          <svg
            class="CircularProgressbar "
            data-test-id="CircularProgressbar"
            viewBox="0 0 100 100"
          >
            <path
              class="CircularProgressbar-trail"
              d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
              fill-opacity="0"
              stroke-width="8"
              style="stroke: var(--color-bg-secondary); stroke-linecap: butt; transform: rotate(0.625turn); transform-origin: center center; stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 72.25663103256524px;"
            />
            <path
              class="CircularProgressbar-path"
              d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
              fill-opacity="0"
              stroke-width="8"
              style="stroke: var(--color-primary); stroke-linecap: butt; transform: rotate(0.625turn); transform-origin: center center; transition-duration: 0.5s; stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 126.44910430698917px;"
            />
          </svg>
          <div
            data-test-id="CircularProgressbarWithChildren__children"
            style="position: absolute; width: 100%; height: 100%; margin-top: -100%; display: flex; flex-direction: column; justify-content: center; align-items: center;"
          >
            <div
              class="content"
            >
              <div
                class="title"
              >
                500
                <span>
                  kcal
                </span>
              </div>
              <div
                class="subtitle"
              >
                Sisa
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
